packages/webview_flutter/webview_flutter_android/lib/src/android_webview.dart

```diff
@@ -766,6 +768,12 @@ class WebViewClient extends JavaObject {
   /// reflect the state of the DOM at this point.
   final void Function(WebView webView, String url)? onPageFinished;
 
+  /// Notify the host application that an HTTP error has been received from the
+  /// server while loading a resource. HTTP errors have status codes >= 400.
+  /// This callback will be called for any resource (iframe, image, etc.), not just for the main page.
+  /// Thus, it is recommended to perform minimum required work in this callback.
+  final void Function(WebView webView, int statusCode)? onPageError;
```

Comment:
This should be called `onReceivedHttpError` which matches the name in the Android class: https://developer.android.com/reference/android/webkit/WebViewClient#onReceivedHttpError(android.webkit.WebView,%20android.webkit.WebResourceRequest,%20android.webkit.WebResourceResponse).

It is also missing the `WebResourceRequest` and `WebResourceResponse` classes. They should replace the `statusCode` parameter and follow the same signature as the Java method. The `WebResourceRequest` class is already implemented and you can use that as an example for the `WebResourceResponse` class. And it's fine to just include the status code for `WebResourceResponse`. The rest of the fields can be added lateer.